garden_ai/app/pipeline.py

```diff
@@ -0,0 +1,207 @@
+import logging
+import re
+from datetime import datetime
+from keyword import iskeyword
+from pathlib import Path
+from typing import List, Optional
+
+import jinja2
+import rich
+import typer
+from garden_ai import GardenClient, Pipeline, step
+from rich import print
+from rich.prompt import Prompt
+
+logger = logging.getLogger()
+
+pipeline_app = typer.Typer(name="pipeline", no_args_is_help=True)
+
+
+def validate_identifier(name: str) -> str:
```

Comment:
(for posterity after sync review) agreed, the word 'validate' was sticky in my brain because of the way pydantic "validators" work similarly (take a maybe-valid value and return a valid one) but that's very opinionated; even if we share pydantic's "parse, don't validate" mindset we don't need to share the "parse, don't validate; also you do the parse with something called `@validator`" mindset https://github.com/pydantic/pydantic/issues/578